### ISSUE TITLE
java.lang.ArrayIndexOutOfBoundsException: 31 on Kevoree MM

### ISSUE THREAD

**Issue by maxleiko** [dukeboard/kevoree-modeling-framework#72]
```java
_metaModel = new org.kevoree.modeling.api.abs.AbstractMetaModel("Kevoree", -1);
        org.kevoree.modeling.api.meta.MetaClass[] tempMetaClasses = new org.kevoree.modeling.api.meta.MetaClass[31];
    // ...
    // DIVISION BY ZERO §
    tempMetaClasses[31] = org.kevoree.meta.MetaNodeType.getInstance();
    // ....
    ((org.kevoree.modeling.api.abs.AbstractMetaModel) _metaModel).init(tempMetaClasses);
```

**Comment by brice-morin** [comment]
Well, I do not know much about KMF implementation details... But it seems normal it crashes since your index is larger than the lengthy of the array... Last element should be tempMetaClasses[30]. But I guess I am overlooking something :-)

**Comment by maxleiko** [comment]
That's why I've added the comment "DIVISION BY ZERO §"

**Comment by brice-morin** [comment]
Yes.... but is that your code? or KMF code? In the first case, you do something wrong, it crashes, so what? In the other case... well... :-)

**Comment by gnain** [comment]
It looks like init code of Meta Element of KMF. Thus generated. It seems to
be linked to the implementation of Enums. We have to check the indexes
indeed.

**Comment by dukeboard** [comment]
I have fixed, release in process, was just a missing exclusion in the brand new ENUM process. Really sorry for that.
I also push a very short example for you on the Kevoree sample in the test directory to ensure the tracking. Will be release into the 4.8.1 version.

**Comment by brice-morin** [comment]
But really, KMF should be tested in details on Kevoree MM.... Kevoree and KMF starts to have some users outside of the Triskell connection, who rapidly get pissed at that kind of bugs...

**Comment by dukeboard** [comment]
Sure that why we added the .mm in our test suite. But really the version 4 of KMF is for many others case studies than Kevoree and that's explains why we focus on more Big Modeling. KMF4 has to be consider as "beta" due to the poor documentation and the instability of the API. We plan to release and advertise on it in the next month when demonstration Java/Js and tutorials will be ready and tested.

**Comment by brice-morin** [comment]
Yes, I agree. But Kevoree should remain an important KMF-powered product :-)

**Comment by dukeboard** [comment]
Of course :-) we plan to re-synchronize soon that's why maxime has start to experiment with it.

**Comment by dukeboard** [comment]
V4.8.1 under propagation in maven repositories

**Comment by maxleiko** [comment]
Fixed in 4.8.1
